finance/server/src/main/java/com/google/finapp/SpannerDaoPGImpl.java

```diff
@@ -0,0 +1,367 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.finapp;
+
+// From SpannerDaoimpl
+import com.google.cloud.ByteArray;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import com.google.cloud.spanner.Value;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ByteString;
+import io.grpc.Status;
+import io.grpc.StatusException;
+import java.math.BigDecimal;
+
+final class SpannerDaoPGImpl implements SpannerDaoInterface {
+
+  private final DatabaseClient databaseClient;
+
+  SpannerDaoPGImpl(DatabaseClient databaseClient) {
+    this.databaseClient = databaseClient;
+  }
+
+  public void createCustomer(ByteArray customerId, String name, String address)
+      throws StatusException {
+    try {
+      this.databaseClient
+          .readWriteTransaction()
+          .run(
+              transaction -> {
+                Statement statement =
+                    Statement.newBuilder(
+                            "INSERT INTO Customer\n"
+                                + "(CustomerId, Name, Address)\n"
+                                + "VALUES\n"
+                                + "(@customerId, @name, @address)")
```

Comment:
nice catch. The named parameters indeed don't work. This wasn't caught because the workload generator doesn't actually create customers (or customer roles). I'll fix this.

Re: Testing, we had discussed adding testing later. Still very important, but given the requirement for an actual spanner instance, that was a secondary task (and thus a later PR).

Reply:
The distinction I wanted to make is:
1. Scripts to run a one-off test VS
2. Automated continuous testing

2 has multiple challenges, but 1 could be more attainable. I'm ok to leave this for a future PR.